lib/ansible/galaxy/data/default/role/meta/main.yml.j2

```diff
@@ -17,22 +20,23 @@ galaxy_info:
   # - CC-BY-4.0
   license: {{ license }}
 
-  min_ansible_version: {{ min_ansible_version }}
-
-  # If this a Container Enabled role, provide the minimum Ansible Container version.
+  # If this a Container Enabled role, provide the minimum Ansible Container
+  # version.
   # min_ansible_container_version:
+  min_ansible_version: "{{ min_ansible_version }}"
```

Comment:
Yes, that seems to be slightly more elegant, I agree.

> The linter rule is wrong, ...

How do you mean that? I am trying to find some guidelines on entries like that, but they are sparse.


The change now should extract the current Ansible version and add it to the variable.
I was thinking first, that 2.1 might be a sane default, but I fail to see that. The previous code just simply cut after the first three characters, så _2.19_ became _2.1_ .. so did _2.18_.

Reply:
> How do you mean that?

It's a restriction `ansible-lint` has created rather than reflecting a guideline/limitation from Galaxy, which is the primary user of the `galaxy_info` dictionary.

> I was thinking first, that 2.1 might be a sane default, but I fail to see that. The previous code just simply cut after the first three characters, så 2.19 became 2.1 .. so did 2.18.

Rather than the regex, this would be a tidier way to get full the major.minor: 

```python
            semantic_version = SemanticVersion.from_loose_version(LooseVersion(ansible_version))
            min_ansible_version = f"{semantic_version.major}.{semantic_version.minor}"
```

And personally, I was in favor of a similar code change, but it was ultimately rejected by the team https://github.com/ansible/ansible/pull/78081. (That was opened before --extra-vars support was added though, so the situation is a little more flexible now.)

Reply:
I agree, this is more elegant.